tests/linearizability/failpoints.go

```diff
@@ -114,45 +123,55 @@ const (
 	Leader    failpointTarget = "Leader"
 )
 
-func (f goFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
-	var member e2e.EtcdProcess
-	switch f.target {
-	case AnyMember:
-		member = clus.Procs[rand.Int()%len(clus.Procs)]
-	case Leader:
-		member = clus.Procs[clus.WaitLeader(t)]
-	default:
-		panic("unknown target")
-	}
+func (f goPanicFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
+	member := f.pickMember(t, clus)
 	address := fmt.Sprintf("127.0.0.1:%d", member.Config().GoFailPort)
-	err := setupGoFailpoint(address, f.failpoint, f.payload)
-	if err != nil {
-		return fmt.Errorf("gofailpoint setup failed: %w", err)
-	}
-	if f.trigger != nil {
-		err = f.trigger(ctx, member)
+
+	triggerCtx, cancel := context.WithTimeout(ctx, triggerTimeout)
+	defer cancel()
+
+	for member.IsRunning() {
+		err := setupGoFailpoint(triggerCtx, address, f.failpoint, "panic")
 		if err != nil {
-			return fmt.Errorf("triggering gofailpoint failed: %w", err)
+			t.Logf("gofailpoint setup failed: %v", err)
+		}
+		if f.trigger != nil {
+			err = f.trigger(triggerCtx, member)
+			if err != nil {
+				t.Logf("triggering gofailpoint failed: %v", err)
+			}
+		}
+		err = member.Wait(triggerCtx)
+		if err != nil && !strings.Contains(err.Error(), "unexpected exit code") {
+			return fmt.Errorf("failed to trigger a process panic within %s, err: %w", triggerTimeout, err)
 		}
 	}
-	err = member.Wait()
-	if err != nil && !strings.Contains(err.Error(), "unexpected exit code") {
-		return err
-	}
-	err = member.Start(ctx)
+
+	err := member.Start(ctx)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func setupGoFailpoint(host, failpoint, payload string) error {
+func (f goPanicFailpoint) pickMember(t *testing.T, clus *e2e.EtcdProcessCluster) e2e.EtcdProcess {
+	switch f.target {
+	case AnyMember:
+		return clus.Procs[rand.Int()%len(clus.Procs)]
```

Comment:
[off topic] Shell we have 3 targets: AnyMember, Leader, NotLeader ? 
Randomly killing either leader or not leader might be a source of flakiness (on the other hand it expands exploratory test coverage).